### PR TITLE
Ticket 4573: Added upgrade step for checking motion set points

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/ISISComputingGroup/IBEX/issues/4573

To test:
* Confirm that all PV name changes made in https://github.com/ISISComputingGroup/EPICS-motionSetPoints/pull/14 are converted in this script
* Confirm that if you have motion set points you are directed to upgrading them

### CODE REVIEW

File: src/upgrade_step_from_7p2p0.py [hunk 1]
```diff
@@ -45,8 +44,66 @@ def perform(self, file_access, logger):
...
+                    " in this version and must be manually fixed".format(MOTION_SET_POINTS_FOLDER)
+                )
+                print(
+                    "See https://github.com/ISISComputingGroup/ibex_developers_manual/"
+                    "wiki/Motion-Set-points#upgrading-from-720 for how to do this"
+                )
+                input("Press any key to confirm this is done.")
+
+            # CoordX:MTR is gone, hard to automatically replace so just raise as issue
+            number_of_motor_refs = changer.change_pv_name("COORD0:MTR", "COORD0:MTR")
```
Comment (JamesKingWork) [inline src/upgrade_step_from_7p2p0.py:97]: A method should do one thing only. I feel like we should have another method that gets the number of occurrences of a pv in blocks and synoptics.

File: src/upgrade_step_from_7p2p0.py [hunk 2]
```diff
@@ -45,8 +44,66 @@ def perform(self, file_access, logger):
...
+            logger (LocalLogger): logger
+
+        Returns: exit code 0 success; anything else fail
+
+        """
+        try:
+            logger.info("Changing motion set point PVs")
+
+            changer = ChangePVsInXML(file_access, logger)
+            num_of_changes = changer.change_pv_name("COORD1", "COORD0")
```
Comment (JamesKingWork) [inline src/upgrade_step_from_7p2p0.py:73]: Unused var num_of_changes

File: src/upgrade_step_from_7p2p0.py [hunk 3]
```diff
@@ -45,8 +44,66 @@ def perform(self, file_access, logger):
...
+                    "wiki/Motion-Set-points#upgrading-from-720 for how to do this"
+                )
+                input("Press any key to confirm this is done.")
+
+            # CoordX:MTR is gone, hard to automatically replace so just raise as issue
+            number_of_motor_refs = changer.change_pv_name("COORD0:MTR", "COORD0:MTR")
+            number_of_motor_refs += changer.change_pv_name("COORD1:MTR", "COORD1:MTR")
+
+            if number_of_motor_refs > 0:
+                print("The PV COORDX:MTR has been found in a config/synoptic but no longer exists")
```
Comment (JamesKingWork) [inline src/upgrade_step_from_7p2p0.py:101]: We are printing out something like "INFO: Writing xml ...". It is not clear to the upgrader which of these are the files containing COORDX:MTR, can we make it more clear which files contain them so as to make it easier to correct them

File: src/upgrade_step_from_7p2p0.py [hunk 4]
```diff
@@ -45,8 +44,66 @@ def perform(self, file_access, logger):
...
+                print("Manually replace with a reference to the underlying axis and rerun the upgrade")
+                return ERROR_CODE
```
Comment (JamesKingWork) [inline src/upgrade_step_from_7p2p0.py:103]: If we have to run this a second time due to detector COORDX:MTR any pvs that were COORD2 and have now be changed to COORD1 will subsequently be changed to COORD0 in the second run

Comment (JamesKingWork) [reply src/upgrade_step_from_7p2p0.py:103]: Could stack up the changes made and undo them if this is found? Or reorder the steps in a clever way

File: src/upgrade_step_from_7p2p0.py [hunk 5]
```diff
@@ -45,8 +44,66 @@ def perform(self, file_access, logger):
...
+
+            return SUCCESS_CODE
 
         except Exception as e:
             logger.error("Unable to perform upgrade, caught error: {}".format(e))
-            return self.ERROR_CODE
```
Comment (JamesKingWork) [inline src/upgrade_step_from_7p2p0.py:52]: If the script has an exception and we then rerun the script a second time COORD2 may have been changed to COORD1 and then in the next run to COORD0

File: src/upgrade_step_from_7p2p0.py (hunk 4 above)
Comment (DominicOram) [reply src/upgrade_step_from_7p2p0.py:103]: Very good spot! Thanks.

**Review by JamesKingWork — APPROVED** [review]

**Pull request merged**